### PR TITLE
Search: do not record invalid queries

### PR DESCRIPTION
**Pull request by humitos**
We are generating lot of errors on Sentry because `query='\x00'`. This commit
skips logging those queries.

Reference: https://sentry.io/organizations/read-the-docs/issues/3066070420/?project=148442

### CODE REVIEW

File: readthedocs/search/tasks.py [hunk 1]
```diff
@@ -161,11 +160,21 @@ def delete_old_search_queries_from_db():
 @app.task(queue='web')
 def record_search_query(project_slug, version_slug, query, total_results, time_string):
     """Record/update a search query for analytics."""
-    if not project_slug or not version_slug or not query:
+
+    log.bind(
+        project_slug=project_slug,
+        version_slug=version_slug,
+    )
```
[the remark this answers is not in the document]
Comment (humitos) [reply readthedocs/search/tasks.py:167]: This is how structulog works (https://github.com/jrobichaud/django-structlog). Every `log` instance is tied to each request. Take a look at https://www.structlog.org/en/stable/getting-started.html#building-ctx

[the remark this answers is not in the document]
Comment (benjaoming) [reply readthedocs/search/tasks.py:167]: Shouldn't it be `log = log.bind` ?

[the remark this answers is not in the document]
Comment (humitos) [reply readthedocs/search/tasks.py:167]: I think that is _required_ when running a normal Python application. However, we also have `django-structlog` and we are installing a middleware from there that I understand that "reset" the logger for each different thread/request: https://django-structlog.readthedocs.io/en/latest/getting_started.html

The same happens with the Celery integration we are using from the same module.

I don't remember from the top of my mind exactly how this works, but I think what I'm saying it's pretty close 😄 . I haven't seen log entries with invalid/inconsistent key/values pairs, so I'd assume it's working fine so far; but maybe we just haven't hit the issue yet.

[the remark this answers is not in the document]
Comment (benjaoming) [reply readthedocs/search/tasks.py:167]: It makes sense that it's something that this middleware would provide :+1: Thanks for a great explanation :+1:

**Pull request closed without merging**